core/app/assets/js/hooks/use-action.js

```diff
@@ -1,9 +1,13 @@
 export default function useAction() {
 	return {
 		backToDashboard: () => {
+			// The split is necessary because `elementorAppConfig.return_url` comes from WP with no parameters.
+			const istargetIdenticalToCurrent = location.href.split( /[&#]/ )[ 0 ] === elementorAppConfig.return_url,
+				isReturnUrlLogin = elementorAppConfig.return_url.includes( elementorAppConfig.login_url );
+
 			if ( window.top === window ) {
 				// Directly - in case that the return_url is the login-page, the target should be the admin-page and not the login-page again.
-				window.top.location = elementorAppConfig.return_url.includes( elementorAppConfig.login_url ) ? elementorAppConfig.admin_url : elementorAppConfig.return_url;
+				window.top.location = istargetIdenticalToCurrent || isReturnUrlLogin ? elementorAppConfig.admin_url : elementorAppConfig.return_url;
```

Comment:
If we always want to go back to the dashboard page then why not use:
`window.top.location = '/wp-admin';`

Reply:
This method is also used for the App's close (x) button.
If you entered an app from the Editor, or from a dashboard page that's different than main page, Product wanted it to send you back to where you came from.

You know what, I will ask product about this.